src/shared/utilities/childProcess.ts

```diff
@@ -278,6 +283,29 @@ export class ChildProcess {
         }
     }
 
+    private registerLifecycleListeners(
+        process: child_process.ChildProcess,
+        errorHandler: (error: Error, forceStop?: boolean) => void,
+        timeout?: Timeout
+    ): void {
+        const pid = process.pid
+        ChildProcess.runningProcesses[pid] = this
+
+        const timeoutListener = timeout?.token.onCancellationRequested(({ type }) => {
+            const message = type == 'cancelled' ? 'Cancelled: ' : 'Timed out: '
+            this.log.verbose(`${message}${this}`)
+            errorHandler(new TimeoutError(type), true)
+        })
+
+        const dispose = () => {
+            timeoutListener?.dispose()
+            delete ChildProcess.runningProcesses[pid]
```

Comment:
We used to have [a lint rule on using the `delete` operator.](https://palantir.github.io/tslint/rules/no-dynamic-delete/) Would it be better to just have `runningProcesses` be a map that we can call `.delete(pid)` on?

Reply:
Oh that's interesting. I've never seen that rule before, I guess it's dangerous if you're using `delete` on an object you don't own since you're literally removing the property key. Can just use a map here.